source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -5,4 +5,77 @@ Access Data From a Cursor
 .. default-domain:: mongodb
 
 Read operations that return multiple documents do not immediately return
-all values matching the query.
+all values matching the query. Because a query can potentially match
+large sets of documents, these operations rely upon an object called a
+:java-docs:`MongoCursor </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html>`.
+
+A ``MongoCursor`` fetches documents in batches to reduce both memory
+consumption and network bandwidth usage. 
+
+The following functions use cursors internally, and return the results of the operations:
+
+- ``collection.find()``
+- ``collection.aggregate()``
+- ``collection.listIndexes()``
+- ``db.listCollectionNames()``
+
+Append ``.cursor()`` to use these functions as cursors.  
```

Comment:
This is a weird phrasing.  I think we need to point out a few things here:

* The initiating methods (find, aggregate) don't actually start any operations on the database.  It's not until the application calls one of the terminal methods that anything happens.
* The terminal methods include: cursor(), but also into() and first(). These latter two can be quite useful.
* The initiating methods implement the Iterable interface, so you can also use forEach to iterate.  This is generally used in preference to forEachRemaining on Iterator.
* The initiating methods implement the Iterable interface, so you can technically use the iterator method to iterate, but you should not do so because Iterator doesn't let you ensure that the underlying cursor is closed if for some reason you stop iterating before the underlying cursor is exhausted.  Instead, you should use the cursor() method, which returns a MongoCursor.  MongoCursor implements Iterable, but also Closeable.  So you can iterate the cursor, and also ensure that the cursor is closed in case of early termination (either in a finally block or a try-with-resources statement)

Reply:
I believe I've addressed each bullet in my latest commit.  Please let me know if I've missed anything and I'll be sure to fix the page.

Reply:
I'm not seeing the first, second, or third bullet points addressed anywhere.

Reply:
The first and second bullet, I mention them in lines 17 and 18.  The third bullet, I have in the warning box.

I didn't want to put your bullet word-for-word, so I took parts of each to incorporate into the page.  Would you like me to take more from each bullet?

Reply:
I'll go through them one by one:

* I think it's important to mention that it's not until a terminal method is executed that a database operation is performed.  If not on this page, then somewhere.
* I would like to see into() and first() described as well, with examples
* I still see forEachRemaining described, and not forEach.  I think forEach should be described, with an example, but not forEachRemaining
* This one is good.  It's the one with the warning box.